resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -746,6 +746,9 @@ private[spark] class Client(
       .filterNot(node => node.getSpec.getUnschedulable != null &&
         node.getSpec.getUnschedulable)
       .flatMap(_.getStatus.getAddresses.asScala)
+      // The list contains hostnames, internal and external IP addresses.
+      // we want only external IP addresses in our list
+      // (https://kubernetes.io/docs/admin/node/#addresses)
```

Comment:
On most systems I see, it is the ExternalIP is accessible. For example, on GKE/GCE, it is not possible to use the hostnames to get access to the cluster by default. If the hostnames were resolvable, they should resolve to the same external IP address in a given cluster, which is why I don't see the point of including both.